src/plugins/hidsim/hidsim.cpp

```diff
@@ -106,8 +106,67 @@
 
 #include <libdrakvuf/libdrakvuf.h>  /* eprint_current_time */
 #include "../private.h"             /* PRINT_DEBUG */
+#include "hid_injection.h"          /* hid_inject */
+#include "gui_monitor.h"            /* gui_reconstruct */
+
 #include "hidsim.h"
 
+/*
+ * Checks, if GUI reconstruction is supported on the system under
+ * investigation
+ */
+bool hidsim::check_platform_support(drakvuf_t drakvuf)
+{
+    win_build_info_t bi;
+
+    {
+        vmi_lock_guard vmi(drakvuf);
+
+        if (!vmi_get_windows_build_info(vmi.vmi, &bi))
+            return false;
+
+    }
+
+    if (bi.version == VMI_OS_WINDOWS_7)
+    {
+
+        PRINT_DEBUG("[HIDSIM] GUI reconstruction supported"
+            "on Windows 7\n");
+        return true;
+    }
+
+    PRINT_DEBUG("[HIDSIM] GUI reconstruction is NOT supported "
+        "on this guest system\n");
+
+    return false;
+}
```

Comment:
Pretty sure there are like 10 not needed newlines in there.

Reply:
Performed a clean up to condense the code.